coredocument/coredocument.go

```diff
@@ -80,27 +80,27 @@ func GetDocumentRootTree(document *coredocumentpb.CoreDocument) (tree *proofs.Do
 	tree = &t
 
 	// The first leave added is the signing_root
-	err = tree.AddLeaf(proofs.LeafNode{Hash: document.SigningRoot, Hashed: true, Property: "signing_root"})
+	err = tree.AddLeaf(proofs.LeafNode{Hash: document.SigningRoot, Hashed: true, Property: proofs.NewProperty("signing_root")})
 	if err != nil {
 		return nil, err
 	}
 	// For every signature we create a LeafNode
 	sigLeafList := make([]proofs.LeafNode, len(document.Signatures)+1)
 	sigLengthNode := proofs.LeafNode{
-		Property: "signatures.length",
+		Property: proofs.NewProperty("signatures.length"),
```

Comment:
Simpler would be to do this:

Before add this line:
```go,
signatureProperty := proofs.NewProperty("signatures")
```

And then use:

```suggestion
		Property: signatureProperty.LengthProp(),
```